python/private/pypi/evaluate_markers.bzl

```diff
@@ -0,0 +1,109 @@
+# Copyright 2024 The Bazel Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""A simple function that evaluates markers using a python interpreter."""
+
+load("//python/private:repo_utils.bzl", "repo_utils")
+load(":pypi_repo_utils.bzl", "pypi_repo_utils")
+
+_SRCS = {
+    Label("@pypi__packaging//:BUILD.bazel"): [
+        Label("@pypi__packaging//:packaging/__init__.py"),
+        Label("@pypi__packaging//:packaging/_elffile.py"),
+        Label("@pypi__packaging//:packaging/_manylinux.py"),
+        Label("@pypi__packaging//:packaging/_musllinux.py"),
+        Label("@pypi__packaging//:packaging/_parser.py"),
+        Label("@pypi__packaging//:packaging/_structures.py"),
+        Label("@pypi__packaging//:packaging/_tokenizer.py"),
+        Label("@pypi__packaging//:packaging/markers.py"),
+        Label("@pypi__packaging//:packaging/metadata.py"),
+        Label("@pypi__packaging//:packaging/requirements.py"),
+        Label("@pypi__packaging//:packaging/specifiers.py"),
+        Label("@pypi__packaging//:packaging/tags.py"),
+        Label("@pypi__packaging//:packaging/utils.py"),
+        Label("@pypi__packaging//:packaging/version.py"),
```

Comment:
I wonder if we need to watch these files here, maybe we should move the watching of the files to the main implementation of the extension?

Reply:
Hm. I'm not sure. I'm not too familiar with what does/doesn't get automatically watched for changes for the bzlmod/repo phase. I know files _outside_ the build system need to be watched (e.g. /usr/bin/python3).

I've just assumed that, if it's addressable by the labels, then bazel will track it appropriately, but never actually checked that behavior.